pkg/routeagent_driver/handlers/kubeproxy/endpoint_handler.go

```diff
@@ -129,6 +140,17 @@ func (kp *SyncHandler) RemoteEndpointUpdated(endpoint *submV1.Endpoint) error {
 func (kp *SyncHandler) RemoteEndpointRemoved(endpoint *submV1.Endpoint) error {
 	kp.syncHandlerMutex.Lock()
 	defer kp.syncHandlerMutex.Unlock()
+
+	lastPorcessedTime, ok := kp.remoteEndpointTimeStamp[endpoint.Name]
+
+	if ok && lastPorcessedTime.After(endpoint.CreationTimestamp.Time) {
+		klog.Infof("Ingoring deleted remote %#v since a later endoint was already"+
+			"processed", endpoint)
+		return nil
```

Comment:
The exact problem that we noticed in OCP deployment was that a stale/old endpoint event coming after an updated event. Here we are skipping the old one, but what about deleting any resources (like iptables, routes etc) which point to the old endpoint? Can you please point me where we will handle it with this change?

Reply:
This code correlates the time stamp between endpoints with the same name. I'm curious what sequence of events can occur for endpoint(s) with the same name that would cause an issue? Can the exact scenario sequence that we're trying to solve be outlined in the PR?

Reply:
@aswinsuryan, thanks for the PR. I see that you are now skipping the endpoint processing if its arriving after the new endpoint. Can you please check that on the old gateway node, the following entries are removed. We are seeing this more often in OCP setup.

1. IP route table 150 entries
2. vx-submariner interface 
3. IP xfrm policy/state rules

Reply:
@sridhargaddam This issue occurs when the old g/w node(machine set) is deleted and a new one is created. So in the remote cluster, the old one does not exist. In the local cluster, g/w node did not change, but it received a stale event. So we need to keep that rules.

Is  that what you meant to do?

Reply:
> This issue occurs when the old g/w node(machine set) is deleted and a new one is created.

Is it possible to add tests for this scenario? I'm thinking of the CII Silver requirement: 

> The project MUST add regression tests to an automated test suite for at least 50% of the bugs fixed within the last six months.

Reply:
@dfarrell07 I am not sure if it can be done in kind,may be removing a worker node and adding it back help.